hazelcast/src/main/java/com/hazelcast/wan/impl/WanReplicationPublisherDelegate.java

```diff
@@ -16,53 +16,160 @@
 
 package com.hazelcast.wan.impl;
 
+import com.hazelcast.internal.services.ServiceNamespace;
+import com.hazelcast.monitor.LocalWanPublisherStats;
+import com.hazelcast.spi.partition.PartitionReplicationEvent;
 import com.hazelcast.wan.WanReplicationEvent;
-import com.hazelcast.wan.WanReplicationEndpoint;
 import com.hazelcast.wan.WanReplicationPublisher;
 
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.hazelcast.util.MapUtil.createHashMap;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
 /**
- * Delegating replication publisher implementation
+ * Delegating WAN replication publisher implementation. This implementation
```

Comment:
This is not an impl of `WanReplicationPublisher` any more. Maybe we can rename it to something like `WanReplicationPublishersContainer`?

Reply:
Got it, will do!